### DISCUSSION TITLE
Finding valid paths internally

### DISCUSSION THREAD

**Discussion by nhpip** [ninenines/cowboy#1581] [Q&A]
Hi,

This is an Elixir app utilizing `Plug.Adapters.Cowboy` so this question may not belong here.

Given a list of dispatchers like:

```
        {"/admin/[...]", Plug.Cowboy.Handler, {CollectionServer.AdminRouter, []}},
        {"/mail/[...]", Plug.Cowboy.Handler, {CollectionServer.Router, []}},
        {"/storage/[...]", Plug.Cowboy.Handler, {CollectionServer.Router, []}},
```
What I would like to do is ask `cowboy` if a path is valid and provisioned without actually invoking any logic.

e.g.

```
cowboy:is_path_provisioned("/users/admin/9ccd7a37-38bb-182f-afd9-28bcd3d09bac/info")
```
Could return something like `{ok, provisioned}` or `{error, not_provisioned}`

Thanks

**Comment by essen** [comment]
You can always call `cowboy_router:execute(#{host => <<"...">>, path => <<"...">>}, #{dispatch => Dispatch})` where `Dispatch` is the dispatch list configured in Cowboy. Now I'm not sure how to retrieve the dispatch list on top of plug.
